src/puppetlabs/puppetdb/reports.clj

```diff
@@ -153,3 +166,24 @@
   [reports :- [report-query-schema]]
   (map report-query->wire-v5
        reports))
+
+(defn dash->underscore-report-keys [report]
+  (->> report
+       utils/dash->underscore-keys
+       (sp/update [:resource_events sp/ALL sp/ALL]
```

Comment:
by my reading of the specter docs, this will narrow down the scope of the transformation as follows:

the report map (post initial dash->underscore-keys)
the :resource_events map
all key-value pairs in the :resource_events map
all individual keys and values in the :resource_events map
Meaning that this will run dashes_underscores on both the keys and values. Is this what you want?